main/pipeline/pipeline.cc

```diff
@@ -137,6 +140,240 @@ unique_ptr<parser::Node> runParser(core::GlobalState &gs, core::FileRef file, co
     return nodes;
 }
 
+core::LocOffsets locOffset(pm_location_t *loc, pm_parser_t *parser) {
+    uint32_t locStart = static_cast<uint32_t>(loc->start - parser->start);
+    uint32_t locEnd = static_cast<uint32_t>(loc->end - parser->start);
+
+    std::cout << "locStart: " << locStart << std::endl;
+    std::cout << "locEnd: " << locEnd << std::endl;
+
+    return core::LocOffsets{locStart, locEnd};
+}
+
+unique_ptr<parser::Node> convertPrismToSorbet(pm_node_t *node, pm_parser_t *parser, core::GlobalState &gs) {
+    switch (PM_NODE_TYPE(node)) {
+        case PM_FLOAT_NODE: {
+            pm_float_node *floatNode = (pm_float_node *)node;
```

Comment:
You should use a C++ style cast, in this case, I think `reinterpret_cast` is the correct one, but this is one of those things I need to re-learn everytime I read it :D

```suggestion
            auto floatNode = reinterpret_cast<pm_float_node *>node;
```

Same goes for below